src/main/java/com/minecolonies/entity/ai/citizen/builder/EntityAIStructureBuilder.java

```diff
@@ -444,55 +475,67 @@ private AIState clearStep()
 
     private AIState requestMaterials()
     {
-        if(job.getStructure() == null)
+        if (job.getStructure() == null)
         {
             //fix for bad structures
             job.complete();
         }
 
-        //todo as soon as material handling has been implemented this should be set to work!
         //We need to deal with materials
-        if (!Configurations.builderInfiniteResources)
+        if (!Configurations.builderInfiniteResources && !job.getWorkOrder().isRequested())
         {
-            //TODO thread this
             while (job.getStructure().findNextBlock())
             {
                 if (job.getStructure().doesStructureBlockEqualWorldBlock())
                 {
                     continue;
                 }
 
-                @Nullable Block block = job.getStructure().getBlock();
-                @NotNull ItemStack itemstack = new ItemStack(block, 1);
+                @Nullable final Block block = job.getStructure().getBlock();
+
+                @NotNull final ItemStack itemstack = new ItemStack(block, 1);
 
-                Block worldBlock = BlockPosUtil.getBlock(world, job.getStructure().getBlockPosition());
+                final Block worldBlock = BlockPosUtil.getBlock(world, job.getStructure().getBlockPosition());
 
-                if (itemstack.getItem() != null
-                      && block != null
+                if (block != null
                       && block != Blocks.AIR
                       && worldBlock != Blocks.BEDROCK
                       && !(worldBlock instanceof AbstractBlockHut)
-                      && !isBlockFree(block, 0)
-                      && checkOrRequestItems(new ItemStack(block)))
+                      && !isBlockFree(block, 0))
                 {
-                    job.getStructure().reset();
-                    return this.getState();
+                    @NotNull final IBlockState blockState = job.getStructure().getBlockState();;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this empty statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)